impl/pkg/service/pkarr.go

```diff
@@ -36,30 +39,37 @@ func NewPKARRService(cfg *config.Config, db *storage.Storage) (*PKARRService, er
 	if err != nil {
 		return nil, util.LoggingErrorMsg(err, "failed to instantiate dht")
 	}
+
+	// create and start cache and scheduler
+	ttl := time.Duration(cfg.PkarrConfig.CacheTTLMinutes) * time.Minute
+	cache := ttlcache.New[string, storage.PkarrRecord](
+		ttlcache.WithTTL[string, storage.PkarrRecord](ttl),
+	)
 	scheduler := dhtint.NewScheduler()
-	service := PKARRService{
+	service := PkarrService{
 		cfg:       cfg,
 		db:        db,
 		dht:       d,
+		cache:     cache,
 		scheduler: &scheduler,
 	}
-	if err = scheduler.Schedule(cfg.PKARRConfig.RepublishCRON, service.republish); err != nil {
+	go cache.Start()
```

Comment:
Where/how does this goroutine get terminated? Would suggest that during shutdown of the server, you call `Stop()`

Reply:
I can add a shutdown method, but it should not matter since it's in memory anyway

Reply:
It's good practice so that graceful shutdown happens. This isn't important unless your service is under heavy load.